### ISSUE TITLE
Is wifi calling supported?

### ISSUE THREAD

**Issue by mohammad-aliB** [Jigsaw-Code/outline-client#288]
Hi, I was wondering if this IOS shadowsocks client would vpn the wifi calling system service?

**Comment by trevj** [comment]
@mohammad-aliB Thanks for trying Outline. I have no idea! @alalamav Do you know the answer to this?

**Comment by alalamav** [comment]
I haven't tested this, but it is my understanding that WiFi calling should be routed through the VPN.

**Comment by mohammad-aliB** [comment]
I haven’t tried with this perticular VPN client but in my expirence some VPN clients will route wifi calling through a VPN and some won’t. Any chance you could test it then get back to me?

**Comment by alalamav** [comment]
tldr; I believe that WiFi Calling bypasses the iOS Outline client's VPN.

I tested this on an iOS device connected to WiFi on airplane mode (no cellular connectivity). The words "Wi-Fi" appeared in the status bar, next to the carrier.  I then connected the VPN to a server that I control. I verified that traffic was routed through the proxy. I placed a call to a local number and I was able to have a conversation. 
If the call was being routed through the proxy, I hypothesized that stopping the connection from the server should drop the call. However, that was not the case.

I don't understand precisely how WiFi Calling works, but it seems like the OS is bypassing the VPN for this type of traffic. We need to investigate if it would be possible to intercept the call in the VPN and if other apps out there support it.

**Comment by mohammad-aliB** [comment]
My school blocks the wifi calling ports and/or protocol and I can confirm that the VPN app “hotspotvpn” allows me to get around the block

**Comment by trevj** [comment]
I'm going to close this due to inactivity - @shaomeng thanks for giving us some confidence that Outline does actually route wifi calls.
